### PR TITLE
ref(metrics): Remove debounce_delay aggregator option

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR removes the `debounce_delay` aggregator option since this was not used anymore because of https://github.com/getsentry/relay/pull/3661 and https://github.com/getsentry/relay/pull/3726.

#skip-changelog

### CODE REVIEW

File: relay-metrics/src/aggregatorservice.rs [hunk 1]
```diff
@@ -33,23 +33,9 @@ pub struct AggregatorServiceConfig {
 
     /// The initial delay in seconds to wait before flushing a bucket.
     ///
-    /// Defaults to `30` seconds. Before sending an aggregated bucket, this is the time Relay waits
-    /// for buckets that are being reported in real time. This should be higher than the
-    /// `debounce_delay`.
-    ///
```
Comment (jjbayer) [inline relay-metrics/src/aggregatorservice.rs:39]: Why did you remove this comment?

**Review by jjbayer — APPROVED** [review]

**Review by Dav1dde — APPROVED** [review]

File: relay-metrics/src/aggregator.rs [hunk 2]
```diff
@@ -506,7 +488,7 @@ fn get_flush_time(
         let floored_timestamp = (now.as_secs() / config.bucket_interval) * config.bucket_interval;
         UnixTimestamp::from_secs(floored_timestamp)
             + config.bucket_interval()
-            + config.debounce_delay()
+            + config.initial_delay()
```
Comment (iker-barriocanal) [inline relay-metrics/src/aggregator.rs:491]: This is changing from 10 -> 30, is this intended?

Comment (iambriccardo) [reply relay-metrics/src/aggregator.rs:491]: Yes, it will change only for processing relays since we had `debounce_delay` set to 30 for pop relays.
